lib/dns.js

```diff
@@ -61,17 +61,16 @@ function makeAsync(callback) {
       // The API already returned, we can invoke the callback immediately.
       callback.apply(null, arguments);
     } else {
-      process.nextTick(callMakeAsyncCbNT, callback, arguments);
+      var args = new Array(arguments.length + 1);
+      args[0] = callback;
```

Comment:
It's faster to specify the length up-front since v8 then knows how much to (pre-)allocate.

[Here](http://jsperf.com/assignment-vs-push/14) is a jsperf showing the difference for a short array.